Source/ConversationV1/Tests/ConversationTests.swift

```diff
@@ -0,0 +1,215 @@
+        sendMessageWithContext()
+        
+        // verify error conditions
+        executeNegativeTests()
+    }
+
+    /// Test the Conversation service by executing each operation with invalid parameters.
+    func executeNegativeTests() {
+
+    }
+
+    // MARK: - Helper functions
+
+    /// Load credentials and instantiate Dialog service
+    override func setUp() {
+        super.setUp()
+
+        // identify credentials file
+        let bundle = NSBundle(forClass: self.dynamicType)
+        guard let url = bundle.pathForResource("Credentials", ofType: "plist") else {
+            XCTFail("Unable to locate credentials file.")
+            return
+        }
+
+        // load credentials from file
+        let dict = NSDictionary(contentsOfFile: url)
+        guard let credentials = dict as? Dictionary<String, String> else {
+            XCTFail("Unable to read credentials file.")
+            return
+        }
+
+        // read Dialog username
+        guard let username = credentials["ConversationUsername"] else {
+            XCTFail("Unable to read Conversation username.")
+            return
+        }
+
+        // read Dialog password
+        guard let password = credentials["ConversationPassword"] else {
+            XCTFail("Unable to read Conversation password.")
+            return
+        }
+
+        // instantiate the service
+        if service == nil {
+            service = Conversation(username: username, password: password)
+        }
+    }
+
+    /// Wait for an expectation to be fulfilled.
+    func waitForExpectation() {
+        waitForExpectationsWithTimeout(timeout) { error in
+            XCTAssertNil(error, "Timeout.")
+        }
+    }
+
+    // MARK: - Positive tests
+
+    /// Create the Dialog application
+    func sendInitialMessageNoContext() {
+        let description = "Sending first message with no context."
+        let expectation = expectationWithDescription(description)
+
+        sendInitialMessage() {
+            (response) in
+                expectation.fulfill()
+            }
+
+        waitForExpectation()
+    }
+    
+    /// Create the Dialog application
+    func sendMessageWithContext() {
+        let description = "Sending first message with no context."
+        let expectation = expectationWithDescription(description)
+        
+        sendInitialMessage() {
+            (response) in
+            self.sendFollowupMessage(response) {
+                (followupResponse) in
+                expectation.fulfill()
+            }
+        }
+        
+        waitForExpectation()
+    }
+    
+    private func sendInitialMessage(completion: (result: MessageResponse) -> Void) {
+        service.sendText(
+            initialMessage,
+            context: nil,
+            workspaceID: workspaceID,
+            failure: { (error) in
+                XCTAssertNil(error)
+        }) { (response) in
+            XCTAssertNotNil(response)
```

Comment:
Ah, I see you are verifying the response values here. Again, I think these tests would be easier to read if the code within this function was called directly within the test instead of being hidden in a helper function.